### PR TITLE
Wacomtablet kcm

### PR DESCRIPTION
**Pull request by Thra11**
###### Motivation for this change
Add new package for KDE Wacom Tablet Configuration GUI and optionally include it in the plasma5 package set if `services.xserver.wacom.enable` is true.

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [x] Tested compilation of all pkgs that depend on this change using `nix-shell -p nix-review --run "nix-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

###### Notify maintainers (of related packages)

cc @ttuegel 
cc @abbradar 
cc @cillianderoiste

### CODE REVIEW

File: pkgs/tools/misc/wacomtablet-kcm/default.nix [hunk 1]
```diff
@@ -0,0 +1,32 @@
+{
```
Comment (tilpner) [inline pkgs/tools/misc/wacomtablet-kcm/default.nix]: Commonly no newline here

File: pkgs/tools/misc/wacomtablet-kcm/default.nix [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+  stdenv, mkDerivation, fetchurl, extra-cmake-modules, qtx11extras,
+  plasma-workspace, libwacom, xf86_input_wacom
+}:
+
+mkDerivation {
+  name = "wacomtablet-kcm";
```
Comment (tilpner) [inline pkgs/tools/misc/wacomtablet-kcm/default.nix]: Use pname instead of name here

```suggestion
  pname = "wacomtablet-kcm";
```

File: pkgs/tools/misc/wacomtablet-kcm/default.nix [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+  version = "3.2.0";
+  src = fetchurl {
+    url = "mirror://kde/stable/wacomtablet/3.2.0/wacomtablet-3.2.0.tar.xz";
+    sha256 = "197pwpl87gqlnza36bp68jvw8ww25znk08acmi8bpz7n84xfc368";
+    name = "wacomtablet.3.2.0.tar.xz";
```
Comment (tilpner) [inline pkgs/tools/misc/wacomtablet-kcm/default.nix]: Why is this necessary?

File: pkgs/tools/misc/wacomtablet-kcm/default.nix [hunk 4]
```diff
@@ -0,0 +1,32 @@
+{
+  stdenv, mkDerivation, fetchurl, extra-cmake-modules, qtx11extras,
```
Comment (tilpner) [inline pkgs/tools/misc/wacomtablet-kcm/default.nix]: You don't actually use stdenv, except for meta. Perhaps expect lib here and use that directly for meta

File: pkgs/tools/misc/wacomtablet-kcm/default.nix (hunk 3 above)
Comment (Thra11) [reply pkgs/tools/misc/wacomtablet-kcm/default.nix]: The `name` bit? It probably isn't necessary, just got copied and pasted from something which did require it.

Comment (tilpner) [reply pkgs/tools/misc/wacomtablet-kcm/default.nix]: I think it should be removed if it's unnecessary. It adds another copy of the name and version

File: pkgs/tools/misc/wacomtablet-kcm/default.nix [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+  plasma-workspace, libwacom, xf86_input_wacom
+}:
+
+mkDerivation {
+  name = "wacomtablet-kcm";
+  version = "3.2.0";
+  src = fetchurl {
+    url = "mirror://kde/stable/wacomtablet/3.2.0/wacomtablet-3.2.0.tar.xz";
```
Comment (tilpner) [inline pkgs/tools/misc/wacomtablet-kcm/default.nix]:
```suggestion
    url = "mirror://kde/stable/${pname}/{$version}/${pname}-${version}.tar.xz";
```

(needs `mkDerivation rec {`)

**Comment by tilpner** [comment]
@GrahamcOfBorg build wacomtablet-kcm

File: pkgs/tools/misc/wacomtablet-kcm/default.nix (hunk 5 above)
Comment (Thra11) [reply pkgs/tools/misc/wacomtablet-kcm/default.nix]: This doesn't currently work as `name`/`pname` is `wacomtablet-kcm`, while the upstream source is named `wacomtablet`. I appended "kcm" to the package name as I thought it would be clearer in the context of nixpkgs, but I could rename it "wacomtablet" to match upstream.

Comment (tilpner) [reply pkgs/tools/misc/wacomtablet-kcm/default.nix]: What does "kcm" mean? You can still use version

Comment (Thra11) [reply pkgs/tools/misc/wacomtablet-kcm/default.nix]: KDE Configuration Module. Each part of the KDE system settings is a kcm. Most of them are bundled in the core plasma5 packages, but there are a few individually packaged ones such as `sddm-kcm` and now this one. There are also a couple of other packages which include a kcm alongside the thing which it configures.

Comment (Thra11) [reply pkgs/tools/misc/wacomtablet-kcm/default.nix]: I'm thinking I'll probably drop the `-kcm` and name it `wacomtablet` like upstream. It should be clear enough from the description that it's the KDE config GUI.

File: pkgs/tools/misc/wacomtablet-kcm/default.nix (hunk 1 above)
Comment (Thra11) [reply pkgs/tools/misc/wacomtablet-kcm/default.nix]: Fixed.

File: pkgs/tools/misc/wacomtablet-kcm/default.nix (hunk 5 above)
Comment (Thra11) [reply pkgs/tools/misc/wacomtablet-kcm/default.nix]: I've changed the package name to `wacomtablet` and used `pname` and `version` in the url. Thanks!

**Comment by aanderse** [comment]
If you can't find someone to test let me know and I'll dig my tablet out.

**Comment by Thra11** [comment]
@aanderse If you have time, that would be most appreciated.

**Review by aanderse — APPROVED** [review]
Testing steps taken:
- paste contents of `pkgs/tools/misc/wacomtablet/default.nix` from this PR into `~/wacom.nix`
- add `(pkgs.libsForQt5.callPackage /home/aaron/wacom.nix { })` to `environment.systemPackages` in `configuration.nix`
- add `services.xserver.wacom.enable = true;` to `configuration.nix`
- type `sudo nixos-rebuild test`
- restart x
- observe working settings manager as well as wacom tablet

**Comment by aanderse** [comment]
I didn't look into it... but what is with the "No attempt" and error from the bot?

**Comment by tilpner** [comment]
@GrahamcOfBorg build wacomtablet-kcm
